src/bin/pg_dump/dump_babel_utils.c

```diff
@@ -828,15 +944,14 @@ addFromClauseForLogicalDatabaseDump(PQExpBuffer buf, TableInfo *tbinfo, bool is_
 	else if(strcmp(tbinfo->dobj.name, "babelfish_authid_user_ext") == 0)
 	{
 		appendPQExpBuffer(buf, " FROM ONLY %s a "
-						  "WHERE a.rolname IN ('dbo', 'db_owner', '%s_dbo', '%s_db_owner', '%s_guest') ",
-						  fmtQualifiedDumpable(tbinfo), escaped_bbf_db_name, escaped_bbf_db_name,
-						  escaped_bbf_db_name);
-		/*
-		 * Builtin db users will already be present in the target
-		 * server so no need to dump their catalog data.
-		 */
-		if (is_builtin_db)
-			appendPQExpBufferStr(buf, "LIMIT 0 ");
+						  "INNER JOIN sys.babelfish_sysdatabases b "
+						  "ON a.database_name = b.name COLLATE \"C\" "
+						  "WHERE b.dbid = %d "
+						  "AND a.rolname NOT IN ('dbo', 'db_owner', "
```

Comment:
Why 'dbo' and 'db_owner' ? We shouldn't be including that. If it is needed then it should be migration mode dependent. because in single-db it will work, but on multi-db it won't.

Reply:
Great find, thanks for pointing it out. Actually, now I realized that `dbo` and `db_owner` shouldn't be included here since we do want to dump them as they are object of a user database (single-db). The only objects we don't want to dump (we want to include here) are the ones related to builtin databases (master/msdb/tempdb).